File: 1475.py
number = input()
count = 1
_set = [1 for i in range(10)]
for i in range(len(number)):
    index = int(number[i])
    if index == 6 or index == 9:
        if _set[6] > 0:
            _set[6] -= 1
        elif _set[9] > 0:
            _set[9] -= 1
        else:
            for i in range(10):
                _set[i] += 1
            _set[index] -= 1
            count += 1
    else:
        if _set[index] > 0:
            _set[index] -= 1
        else:
            for i in range(10):
                _set[i] += 1
            _set[index] -= 1
            count += 1
print(count)

File: 1475.c
#include <stdio.h>

int main()
{
    int number, index;
    int count = 1;
    int array[10] = {
        1,
    };
    scanf("%d", &number);
    do
    {
        index = number % 10;
        if (index == 6 || index == 9)
        {
            if (array[6] > 0)
            {
                --array[6];
            }
            else if (array[9] > 0)
            {
                --array[9];
            }
            else
            {
                for (int i = 0; i < 10; ++i)
                {
                    array[i] += 1;
                }
                --array[index];
                ++count;
            }
        }
        else
        {
            if (array[index] > 0)
            {
                --array[index];
            }
            else
            {
                for (int i = 0; i < 10; ++i)
                {
                    array[i] += 1;
                }
                --array[index];
                ++count;
            }
        }

    } while ((number = number / 10) != 0);
    printf("%d\n", count);
    return 0;
}